converter/src/main/java/gov/cms/qpp/conversion/encode/ClinicalDocumentEncoder.java

```diff
@@ -64,14 +65,13 @@ private void encodeToplevel(JsonWrapper wrapper, Node thisNode) {
 				thisNode.getValue(ClinicalDocumentDecoder.TAX_PAYER_IDENTIFICATION_NUMBER));
 		}
 
-		if (ClinicalDocumentDecoder.ENTITY_APM.equals(entityType)) {
-			wrapper.put(ClinicalDocumentDecoder.ENTITY_ID,
-				thisNode.getValue(ClinicalDocumentDecoder.PRACTICE_ID));
+		if (Program.isCpc(thisNode)) {
+			wrapper.put(ClinicalDocumentDecoder.ENTITY_ID, thisNode.getValue(ClinicalDocumentDecoder.PRACTICE_ID));
 		}
 
-		if (ClinicalDocumentDecoder.ENTITY_VIRTUAL_GROUP.equals(entityType)) {
-			wrapper.put(ClinicalDocumentDecoder.ENTITY_ID,
-				thisNode.getValue(ClinicalDocumentDecoder.ENTITY_ID));
+		if (ClinicalDocumentDecoder.ENTITY_VIRTUAL_GROUP.equals(entityType) ||
+			(!Program.isCpc(thisNode) && ClinicalDocumentDecoder.ENTITY_APM.equalsIgnoreCase(entityType))) {
```

Comment:
Is Virtual Groups submitting under the cpc+ program name? I've seen that under Groups, but haven't observed it for VG yet.

Reply:
This is for MIPS APM's to be able to have a practiceId for both virtual groups and MIPS apms as I believe it is required for both.

Reply:
There is code that uses the same identifier label for APM/CPC+ Practice id's and uses it for virtual groups. It maps the name and value like so 'entityId' : 'virtualGroupIdentifier'. I think because these 3 program types have similar identifiers encoded into QPP format, it got a little mixed up.